src/main/ipc.ts

```diff
@@ -20,6 +20,14 @@ ipcMain.handle('procedures:getProcedure', async (_event, ...args) => {
   return new Procedures().fetchContent(args[0]);
 });
 
+ipcMain.handle('procedures:checkTableExists', async (_event, ...args) => {
+  return new Procedures().checkTableExists(args[0]);
+});
+
+ipcMain.handle('procedures:numRecordsInTable', async (_event, ...args) => {
+  return new Procedures().numRecordsInTable(args[0]);
+});
+
```

Comment:
```suggestion
```

I don't think these functions need to go in the ipc renderer. The front end will just be passing the RuleType model to the back end and then based on the fields within, the backend will choose the correct function to call and return a response from (as you are doing in pass-fail)

Reply:
Yeah makes sense